app/services/workers/process-import.js

```diff
@@ -15,35 +17,81 @@ const submittingAgent = require('../../constants/task-submitting-agent')
 
 module.exports.execute = function (task) {
   const batchSize = parseInt(config.ASYNC_WORKER_BATCH_SIZE, 10)
+  var tasks = []
+  var workloadReportId
 
-  return getExtractIdRange().then(function (idRange) {
-    var tasks = []
-    var numberOfRecordsToProcess = idRange.lastId - idRange.firstId
-    var tasksRequired = Math.ceil(numberOfRecordsToProcess / batchSize)
-
-    logger.info('Creating ' + tasksRequired + ' create workload tasks')
-
-    if (tasksRequired > 0) {
-      var nextId = idRange.firstId
-      insertWorkloadReport().then(function (workloadReportId) {
-        for (var i = 0; i < tasksRequired; i++) {
-          var additionalData = new Batch(nextId, batchSize)
-          var taskToWrite = new Task(
-            undefined,
-            submittingAgent.WORKER,
-            taskType.CREATE_WORKLOAD,
-            additionalData,
-            workloadReportId,
-            undefined,
-            taskStatus.PENDING
-          )
-          tasks.push(taskToWrite)
-          nextId += batchSize
-        }
-        return createNewTasks(tasks)
+  return insertWorkloadReport()
+  .then(function (insertedWorkloadReportId) {
+    workloadReportId = insertedWorkloadReportId
+    return populateStagingCourtReporters()
+  })
+  .then(function () {
+    return getCourtReportsTasks(tasks, batchSize, workloadReportId)
+  })
+  .then(function (tasks) {
+    return getWorkloadTasks(tasks, batchSize, workloadReportId)
+  })
+  .then(function (tasks) {
+    if (tasks.length > 0) {
+      return createNewTasks(tasks)
+      .then(function () {
+        logger.info('Tasks created')
       })
     }
+  })
+}
 
-    logger.info('Tasks created')
+var populateStagingCourtReporters = function () {
+  return getCourtReportsWithNoWorkloads()
+  .then(function (courtReports) {
+    return replaceCourtReporters(courtReports)
   })
 }
+
+var getCourtReportsTasks = function (tasks, batchSize, workloadReportId) {
```

Comment:
This will create and get the court report tasks - can we update the method name to reflect that?